test/hummingbot/connector/exchange/bitmart/test_bitmart_api_order_book_data_source.py

```diff
@@ -362,3 +327,52 @@ def test_websocket_connection_creation_raises_exception_after_loging(self, mock_
             asyncio.get_event_loop().run_until_complete(self.data_source._create_websocket_connection())
 
         self.assertTrue(self._is_logged("NETWORK", "Unexpected error occurred during bitmart WebSocket Connection ()"))
+
+    def _trade_ws_messsage(self):
+        resp = {
+            "table": "spot/trade",
+            "data": [
+                {
+                    "symbol": "ETH_USDT",
+                    "price": "162.12",
+                    "side": "buy",
+                    "size": "11.085",
+                    "s_t": 1542337219
+                },
+                {
+                    "symbol": "ETH_USDT",
+                    "price": "163.12",
+                    "side": "buy",
+                    "size": "15",
+                    "s_t": 1542337238
+                }
+            ]
+        }
+        return ujson.dumps(resp)
+
+    @patch('websockets.connect', new_callable=AsyncMock)
+    def test_listen_for_trades(self, mock_ws):
+        msg_queue: asyncio.Queue = asyncio.Queue()
+        mock_ws.return_value = self.mocking_assistant.create_websocket_mock()
+        mock_ws.close.return_value = None
+
+        # Add message to be processed after subscriptions, to unlock the test
+        self.mocking_assistant.add_websocket_text_message(mock_ws.return_value, self._trade_ws_messsage())
+        BitmartAPIOrderBookDataSource._trading_pairs = ["ETH-USDT"]
+
+        listening_task = self.ev_loop.create_task(
+            self.data_source.listen_for_trades(self.ev_loop, msg_queue)
+        )
+        trade1: OrderBookMessage = self.ev_loop.run_until_complete(msg_queue.get())
+        trade2: OrderBookMessage = self.ev_loop.run_until_complete(msg_queue.get())
+
+        try:
+            listening_task.cancel()
+            asyncio.get_event_loop().run_until_complete(listening_task)
+        except asyncio.CancelledError:
+            # The exception will happen when cancelling the task
+            pass
```

Comment:
There is no need for this, since the tearDown() takes care of cancelling the listening_task

Reply:
Good point, didn't realise that.